plutus-chain-index/src/Plutus/ChainIndex/UtxoState.hs

```diff
@@ -159,24 +169,23 @@ data RollbackResult a =
         , rolledBackIndex :: UtxoIndex a
         }
 
-viewTip :: Measured (UtxoState a) (UtxoState a)
-        => UtxoIndex a
-        -> Tip
-viewTip = tip . measure
-
 -- | Perform a rollback on the utxo index
 rollback :: Point
          -> UtxoIndex TxUtxoBalance
          -> Either RollbackFailed (RollbackResult TxUtxoBalance)
+rollback PointAtGenesis (viewTip -> TipAtGenesis) = Right (RollbackResult TipAtGenesis mempty)
 rollback _ (viewTip -> TipAtGenesis) = Left RollbackNoTip
 rollback targetPoint idx@(viewTip -> currentTip)
+    -- Already at the target point
```

Comment:
When resuming from a point, the cardano chain sync service always sends a rollback command from that point, even if the point is the current tip. From that it's clear that a rollback to the current tip should be allowed. Should this also be done in the rollback function in TxIdState? @j-mueller @silky

Reply:
Yes I believe so. (Ideally we wouldn't even have two separate rollback functions, that could be refactored if inspiration strikes you.)